File: ScriptTester/src/main/scala/APIObject.scala
object APIObject {

  def instance() = this
  def note(theNote: String) = System.out.println(theNote)
  
  def download(filePath: String): String =  {
    System.out.println("APIObject Downloading: " + filePath)
    Utilities.download(filePath)
  }

  def readFile(absolutePath: String): String = {
    System.out.println("APIObject Reader Path is: " + absolutePath)
    try {
      return Utilities.readFile(absolutePath)
    } catch {
      case e:Exception => System.out.println("Could not open in APIObject:readFile" + e.toString)
    }
    "FAILURE"
  }
  def slice(toSlice: String, a: Int, b: Int): String = toSlice.substring(a,b)
}

File: common/src/main/java/io/githup/limvot/mangagaga/APIObject.kt
package io.githup.limvot.mangagaga

import java.io.File;
import java.lang.Thread;

object APIObject : GenericLogger {
  fun instance() = this
  fun note(theNote: String) = info(theNote)
  // This lambda is reassigned by ImageViewerActivity to be a title bar change
  var onStatus = { text:String -> note("status: $text") }
  fun status(text: String) = onStatus(text)
  
  fun download(filePath: String): String {
    info("Downloading $filePath")
    return Utilities.download(filePath)
  }

  fun downloadWithRequestHeadersAndReferrer(filePath: String, referer: String): Pair<String, MutableMap<String,List<String>>> {
    info("Downloading $filePath")
    return Utilities.downloadWithRequestHeadersAndReferrer(filePath, referer)
  }

  fun readFile(absolutePath: String): String {
    try {
      return File(absolutePath).readText()
    } catch (e: Exception) {
      error("Could not open in APIObject:readFile $e")
    }
    return "FAILURE"
  }
  fun sleep(time: Int) = Thread.sleep(time.toLong())
}
